conjure-undertow-lib/src/main/java/com/palantir/conjure/java/undertow/lib/TypeMarker.java

```diff
@@ -57,4 +62,29 @@ public final Type getType() {
     public final String toString() {
         return "TypeMarker{type=" + type + '}';
     }
+
+    /** Create a new {@link TypeMarker} instance wrapping the provided {@link Type}. */
+    public static TypeMarker<?> of(Type type) {
+        return new WrappingTypeMarker(type);
+    }
+
+    private static final class WrappingTypeMarker extends TypeMarker<Object> {
+        private WrappingTypeMarker(Type type) {
+            super(type);
+        }
+
+        @Override
+        public int hashCode() {
+            return getType().hashCode();
+        }
+
+        @Override
+        public boolean equals(Object other) {
+            if (other instanceof WrappingTypeMarker) {
+                WrappingTypeMarker otherMarker = (WrappingTypeMarker) other;
+                return Objects.equals(getType(), otherMarker.getType());
+            }
+            return false;
+        }
```

Comment:
Should these be moved to `TypeMarker`? I don't see any reason why we wouldn't want this implementation for all `TypeMarker` instances.

I guess maybe it's a little weird to consider different anonymous types equal, but that feel like the correct behavior for `TypeMarker`.

Reply:
I went back and forth on this. We definitely want to allow caches to work correctly for these, where in the usual anonymous class creation path we’d get the same instance and didn’t need to worry about it.
Equals and hashcode are pretty special in that they should take implementation details into account, so I don’t think it would be correct to make them final in the base class.

I think in a follow-up id like to update our caches to create a copy of TypeMarker instances used as keys using TypeMarker.of(marker.getType()) to avoid holding onto encapsulating classes

Reply:
https://github.com/palantir/conjure-java/pull/2265